source-http-ingest/src/server.rs

```diff
@@ -375,13 +399,27 @@ impl Handler {
         // impact other requests. So this ensures that each request is all-or-
         // nothing, which is probably simpler for users to reason about anyway.
         for (_, doc) in enhanced_docs.iter() {
-            let validation_result = collection.validator.validate(None, doc)?;
-            if let Err(validation_err) = validation_result.ok() {
-                tracing::info!(error = %validation_err, uri_path = %collection_path, "request document failed validation");
+            let CollectionHandler {
+                schema_index,
+                schema_url,
+                ..
+            } = &collection;
+            let mut validator: Validator<doc::Annotation, json::validator::SpanContext> =
+                Validator::new(schema_index);
+            validator
+                .prepare(schema_url)
+                .context("preparing validator")?;
+
+            let mut deserializer = serde_json::Deserializer::from_str(&doc);
```

Comment:
i don't have a useful suggestion, but noting we are trading a gain in memory efficiency at a cost of parsing the JSON document twice.

In the future we could consider using a doc::Parser and parsing into a doc::HeapNode, which then has its bump allocator freed ASAP.